src/main/java/org/codevillage/InheritanceParsingStep.java

```diff
@@ -0,0 +1,87 @@
+package org.codevillage;
+
+import com.github.javaparser.ast.CompilationUnit;
+import java.util.Set;
+import java.util.HashSet;
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+
+public class InheritanceParsingStep extends EntityParsingStep {
+
+    public InheritanceParsingStep(EntityParsingChain next) {
+        super(next);
+    }
+
+    @Override
+    public JavaEntity construct(EntityBuilder builder, CompilationUnit declaration) {
+        // Existing logic to parse the entity
+        JavaEntity entity = next.construct(builder, declaration);
+
+        // New logic to get associated class names
+        Set<String> associatedClassNames = getAssociatedClassNames(entity);
+        associatedClassNames.forEach(builder::addAssociation);
+
+        return entity;
+    }
+
+    private Set<String> getAssociatedClassNames(Object entity) {
```

Comment:
Isn't your task to just get the parent class of the ClassOrInterfaceDeclaration if it has one? Kind of confused on why all this logic is needed